flink-core/src/test/java/org/apache/flink/testutils/TestFileSystem.java

```diff
@@ -37,28 +43,45 @@
 
     public static final String SCHEME = "test";
 
-    private static int streamOpenCounter;
+    // number of (input) stream opened
+    private static final AtomicInteger streamOpenCounter = new AtomicInteger(0);
+
+    // current number of created, unclosed (output) stream
+    private static final Map<Path, Integer> currentUnclosedOutputStream = new ConcurrentHashMap<>();
 
     public static int getNumtimeStreamOpened() {
-        return streamOpenCounter;
+        return streamOpenCounter.get();
+    }
+
+    public static Map<Path, Integer> getCurrentUnclosedOutputStream() {
+        return currentUnclosedOutputStream;
     }
 
-    public static void resetStreamOpenCounter() {
-        streamOpenCounter = 0;
+    public static void resetStreamCounter() {
```

Comment:
Can we just `reset(Path)`? I think maybe there are some unstable cases if just clear all counters.